### ISSUE TITLE
interaction with payment methods doesn't appear to be defined

### ISSUE THREAD

**Issue by dbaron** [w3c/webpayments-payment-apps-api#249]
The [definition of `PaymentInstrument`](https://w3c.github.io/payment-handler/#paymentinstrument-dictionary) defines a `capabilities` member that is described as:

> The capabilities member is a list of payment-method-specific capabilities that this payment handler is capable of supporting for this instrument. For example, for the basic-card payment method, this object will consist of an object with two fields: one for supportedNetworks, and another for supportedTypes.

This seems to define how it works by example.  I would expect something to say which specification is responsible for the definition for each payment method.  For example, it's common for a specification like this to define a term that each specification for a payment method would then have to reference in order to define how it interacts here.

It's also not clear how `capabilities` interactions with the possibility that `enabledMethods` has more than one member.

(I got here from w3ctag/design-reviews#231.)

**Comment by ianbjacobs** [comment]
Hi @dbaron,

Agreed: we will need to fix the definition so that it is not definition by example. 

In previous discussions it was proposed [1] that we identify the "capability matching"
parts of request data explicitly. Along with various proposals to "standardize" the
expression of capabilities in payment method specs, some hoped that explicit labeling
of filters would make it easier to automate capability matching in future payment method
specs. Those proposals did not gain traction with browser vendors. 

The current expectation to my understanding is this:

1) No URL-identified payment methods have requested capability matching.
2) Multiple W3C payment methods have requested capability matching. That makes sense because the W3C payment methods generally seek to abstract across multiple similar systems, so up-front filtering continues to appear as a way to maximize the changes of a good user experience in matched payment handlers.
3) At least Google has said "We will look at each W3C payment method and decide whether to implement it." Part of the implementation will include capability matching. 
4) The current proposal around payment handlers is that for URL-identified payment methods capability matching will happen in the payment handler, and for W3C-defined payment methods it will happen in the browser.
5) We anticipate, therefore, that browser implementers will know which data is to be used for filtering by reading the payment method specs.

See also: pull request #170 (canMakePaymentEvent)

[1] https://github.com/w3c/webpayments/issues/170

**Comment by adrianhopebailie** [comment]
@ianbjacobs a good topic for next call?

**Comment by rsolomakhin** [comment]
@marcoscaceres: Should the payment handler spec invoke the **can make a payment** algorithm as defined in [your pull request](https://github.com/w3c/payment-method-basic-card/pull/52/files#diff-eacf331f0ffc35d4b482f1d15a887d3bR241) from the [Filtering of payment instruments](https://w3c.github.io/payment-handler/#filtering-of-payment-instruments) section?

**Comment by marcoscaceres** [comment]
@rsolomakhin no, because that's for "basic card" only... in handler, it just needs to explain how a developer would implement the behavior on their own (and respond with "true" or "false"). You could point to Basic Card tho, as an example.

**Comment by ianbjacobs** [comment]
@dbaron and @rsolomakhin, can we close this issue since we now have canMakePayment() instead of capabilities?

**Comment by rsolomakhin** [comment]
@ianbjacobs : We have both `capabilities` and `CanMakePaymentEvent`, if I understand the spec correctly. The `capabilities` are used for standardized payment method identifiers. The `CanMakePaymentEvent` may be used for URL-based payment method identifiers. See [Filtering of Payment Instruments](https://w3c.github.io/payment-handler/#filtering-of-payment-instruments) for more info.

**Comment by ianbjacobs** [comment]
@rsolomakhin,

Very strange. I was looking for capabilities in the spec yesterday and did not see them. I wonder if I was looking in the wrong spec. In any case, you are obviously correct!

I think we have more to say about capabilities now that we have (1) algorithms in payment method specs and (2) filtering of payment methods. I will come back to this with some proposed text (but not immediately).
